frontend/src/index.tsx

```diff
@@ -10,19 +10,43 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import App, { store } from './App';
 import * as serviceWorker from './serviceWorker.ignore';
-import Keycloak, { KeycloakInstance } from 'keycloak-js';
 import { KeycloakProvider } from '@react-keycloak/web';
 import { Provider } from 'react-redux';
 import getKeycloakEventHandler from 'utils/KeycloakEventHandler';
 import { AuthStateContextProvider } from 'contexts/authStateContext';
 import { BrowserRouter as Router } from 'react-router-dom';
 import EmptyLayout from 'layouts/EmptyLayout';
 import LoginLoading from 'features/account/LoginLoading';
+import { createKeycloakInstance } from 'utils';
+import Keycloak from 'keycloak-js';
 
-//@ts-ignore
-const keycloak: KeycloakInstance = new Keycloak('/keycloak.json');
+/**
+ * Displays LoginLoading until Keycloak connection is ready.
+ * @returns Index component.
+ */
 const Index = () => {
-  return (
+  const [loading, setLoading] = React.useState(true);
+  const [keycloak, setKeycloak] = React.useState(Keycloak);
+
+  React.useEffect(() => {
```

Comment:
Provide a way to override the keycloak configuration file locally through environment variables.